R/nm-join.R

```diff
@@ -149,12 +151,12 @@ nm_join <- function(
       # do the join
       tab <- drop_dups(tab, .d, "ID", .n)
       col_order <- union(col_order, names(tab))
-      .d <- join_fun(tab, .d, by = "ID")
+      .d <- join_fun(tab, .d, by = "ID") %>% arrange(ID)
```

Comment:
> Curious why this needs to be arranged?

As in, why call `arrange()` in this spot?  Don't you already answer that in your [other](https://github.com/metrumresearchgroup/bbr/pull/508#discussion_r910152435) [comments](https://github.com/metrumresearchgroup/bbr/pull/508#issuecomment-1170186401), basically "it doesn't need arrange, always use left_merge".

But, since you asked this question, I'm probably missing something (perhaps a distinction between this join on line 154 and the one on line 159).

Reply:
@kyleam - I didn't pick up on the root cause of this at first.  

Sometimes the table output is `FIRSTONLY` ... that is one row per `ID`.  So regardless of `superset` or no `superset`, this is always a `left_join()` with the accumulating data on the left and the `FIRSTONLY` table on the right.

Reply:
Here's the processing for `FIRSTONLY`

```r
    } else if (nrow(tab) == nid) {
      # if FIRSTONLY table join on ID
      verbose_msg(glue("{.n} is FIRSTONLY table"))

      # if ID is missing, get it from the data by using .join_col
      if (!has_id) { 
        tab <- tab %>%
          left_join(select(.d, .data$ID, !!.join_col), by = .join_col) 
      }

      # toss .join_col, if present, because we're joining on ID
      tab[[.join_col]] <- NULL

      # do the join
      tab <- drop_dups(tab, .d, "ID", .n)
      col_order <- union(col_order, names(tab))
      .d <- join_fun(tab, .d, by = "ID") %>% arrange(ID)  # <---- **
```

I think this last join should be

```r
 .d <- left_join(.d, tab, by = "ID") 
```